platform/source/posix/iot_threads_posix.c

```diff
@@ -179,10 +231,69 @@ bool Iot_CreateDetachedThread( IotThreadRoutine_t threadRoutine,
             {
                 IotLogError( "Failed to set detached thread attribute. errno=%d.",
                              posixErrno );
-                ( void ) pthread_attr_destroy( &threadAttributes );
 
                 status = false;
             }
+
+            /* Set the stack size if given. */
+            if( status == true )
+            {
+                if( stackSize != IOT_THREAD_DEFAULT_STACK_SIZE )
+                {
+                    posixErrno = pthread_attr_setstacksize( &threadAttributes,
+                                                            stackSize );
+
+                    if( posixErrno != 0 )
+                    {
+                        IotLogError( "Failed to set stack size %lu.", ( unsigned long ) stackSize );
+
+                        status = false;
+                    }
+                }
+            }
+
+            /* Set scheduler policy and priority if given. */
+            if( status == true )
+            {
+                if( priority != IOT_THREAD_DEFAULT_PRIORITY )
+                {
+                    struct sched_param schedulerParam =
+                    {
+                        .sched_priority = ( int ) priority
+                    };
+
+                    /* Set scheduler policy SCHED_RR. */
+                    posixErrno = pthread_attr_setschedpolicy( &threadAttributes,
+                                                              SCHED_RR );
+
+                    if( posixErrno != 0 )
+                    {
+                        IotLogError( "Failed to set scheduling policy SCHED_RR." );
+
+                        status = false;
+                    }
+
+                    /* Set scheduler priority. */
+                    if( status == true )
+                    {
+                        posixErrno = pthread_attr_setschedparam( &threadAttributes,
+                                                                 &schedulerParam );
+
+                        if( posixErrno != 0 )
+                        {
+                            IotLogError( "Failed to set scheduling priority %d.",
+                                         schedulerParam.sched_priority );
+
+                            status = false;
+                        }
+                    }
+                }
+            }
+
+            if( status == false )
+            {
+                ( void ) pthread_attr_destroy( &threadAttributes );
```

Comment:
I do not know this well enough to understand what this destroy call is doing, the variable is on the stack and will be cleaned up and I did not see any calls which seemed to obviously allocate anything. This may be fine, just consider if this is required.